src/compiler/compile.js

```diff
@@ -716,6 +717,11 @@ function compileDirectives (attrs, options) {
         pushDir(dirName, internalDirectives[dirName])
       } else {
         arg = dirName
+        // support for shorthand expression (#2843)
+        // <div v-bind:content> => <div v-bind:content="content">
+        if (!value) {
```

Comment:
I'm not sure if this is what you mean, but `value` is the expression, not the return value of the expression. So you can still do `v-bind:content="null"`.